docs/tasks/administer-cluster/kubeadm-upgrade-ha.md

```diff
@@ -0,0 +1,179 @@
+
+## Upgrading your control plane
+
+The following procedure is applied on all master nodes one by one, i.e. when it has finished on master _#1_ it is applied on master _#2_ etc.
+
+Before the actual upgrade `configmap/kubeadm-config` needs to be adapted - replace a hard reference to some master host by the current master host’s FQDN:
+
+```shell
+$ kubectl get configmap -n kube-system kubeadm-config -o yaml >/tmp/kubeadm-config-cm.yaml
+$ sed -i 's/^\([ \t]*nodeName:\).*/\1 <CURRENT-MASTER-FQDN>/' /tmp/kubeadm-config-cm.yaml
+$ kubectl apply -f /tmp/kubeadm-config-cm.yaml --force
+```
+
+Now the actual upgrade process can start. Use the target version determined in the preparation step and run the following command (press “y” when prompted):
+
+```shell
+$ /tmp/kubeadm upgrade apply v<YOUR-CHOSEN-VERSION-HERE>
+```
+
+If the operation was successful you’ll get a message like this:
+
+    [upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.9.2". Enjoy!
+
+    To upgrade the cluster with CoreDNS as the default internal DNS, invoke `kubeadm upgrade apply` with the `--feature-gates=CoreDNS=true` flag.
+
+`kubeadm upgrade apply` does the following:
+
+- Checks that your cluster is in an upgradeable state:
+  - The API server is reachable,
+  - All nodes are in the `Ready` state
+  - The control plane is healthy
+- Enforces the version skew policies.
+- Makes sure the control plane images are available or available to pull to the machine.
+- Upgrades the control plane components or rollbacks if any of them fails to come up.
+- Applies the new `kube-dns` and `kube-proxy` manifests and enforces that all necessary RBAC rules are created.
+- Creates new certificate and key files of apiserver and backs up old files if they're about to expire in 180 days.
+
+Next, manually upgrade your Software Defined Network (SDN).
+
+Your Container Network Interface (CNI) provider may have its own upgrade instructions to follow. Check the [addons](/docs/concepts/cluster-administration/addons/) page to find your CNI provider and see if there are additional upgrade steps necessary.
+
+It has been observed that in some cases after having upgraded the primary master the upgrade on some of the secondary masters failed initially (timed out waiting for the restarted static pods to come up) but worked just fine when retried after a minute or two.
+
+## Upgrade base software packages
+
+At this point all the static pod manifests in your cluster, i.e. API Server, Controller Manager, Scheduler, Kube Proxy have been upgraded, however the base software, i.e. `kubelet`, `kubectl`, `kubeadm` installed on your nodes’ OS are still of the old version. For upgrading the base software packages on all nodes they will be taken out (`cordon` if master, `drain` if worker node) upgraded and then taken in (`uncordon`) again one by one. 
```

Comment:
replace "i.e." with "for example"

Reply:
I believe it could be considered safe to just refer to cordon, drain, and uncordon directly here, instead of using 'taken out' and 'taken in'. There should also be a comma between the step of cordon/draining the node and "upgraded"

Reply:
Done.